[Test] new GitHub Actions config

daniel-sosna opened the pull request.

daniel-sosna committed:
refactor(CI): split into backend and frontend

daniel-sosna committed:
refactor(CI): extract steps as composite actions

daniel-sosna committed:
refactor(CI): use inputs in composite actions

daniel-sosna committed:
refactor(CI): use dotnet and node built-in caching

daniel-sosna committed:
task(CI): add server tests with coverage

daniel-sosna committed:
fix(CI): GitHub Actions specific syntax errors

Fixes:
- incorrect trigger's paths patterns,
- not specifying shell in composite actions
- mistyped indentation of cache properties

daniel-sosna committed:
chore(CI): omit unnecessary quotes

Follows YAML conventions

rdidrikas commented:
Hey, can you give a quick description on what this does? Will this run tests both on frontend and backend on pushes?

daniel-sosna committed:
task(CI): add coverage upload to Codecov

daniel-sosna committed:
task(CI): add frontend tests with coverage

daniel-sosna committed:
task(CI): add job filter

Improve paths in triggers for better worklows evoking logic
Add changes detection job to run only necessary jobs

daniel-sosna committed:
fix: run dotnet format and fix errors in CI

daniel-sosna committed:
Merge remote-tracking branch 'origin' into task/modular-ci

daniel-sosna committed:
fix: failed CI runs

* Run dotnet format on tests projects
* Add cache-dependency-path into node test action
* Rename button.jsx to Button.jsx

daniel-sosna committed:
Merge branch 'task/modular-ci' of github.com:daniel-sosna/draw-it into task/modular-ci
